frontend/src/components/MainNavigation.vue

```diff
@@ -292,51 +349,93 @@ export default {
         name = fallback
       }
       return !this.namespaced ? { name, query: { namespace } } : { name, params: { namespace } }
-    }
-  },
-  watch: {
-    projectMenu (value) {
-      if (value) {
-        requestAnimationFrame(() => {
-          setDelayedInputFocus(this, 'projectFilter')
-        })
+    },
+    onInputProjectFilter () {
+      this.highlightedProjectName = undefined
+      this.numberOfVisibleProjects = initialVisibleProjects
+      if (this.projectFilterHasExactMatch) {
+        this.highlightedProjectName = this.projectFilter
       }
-    }
-  }
-}
-</script>
+      this.$nextTick(() => this.scrollSelectedProjectIntoView())
+    },
+    selectProjectWithKeys (keyDirection) {
+      let currentHighlightedIndex = this.findProjectIndexCaseInsensitive(this.highlightedProject.metadata.name)
 
-<style lang="styl">
-  .project-menu {
-    border-radius: 0;
+      if (keyDirection === 'up') {
+        if (currentHighlightedIndex > 0) {
+          currentHighlightedIndex--
+        }
+      } else if (keyDirection === 'down') {
+        if (currentHighlightedIndex < this.sortedAndFilteredProjectListWithAllProjects.length - 1) {
+          currentHighlightedIndex++
+        }
+      }
 
-    .v-card {
-      border-radius: 0;
+      const newHighlightedProject = this.sortedAndFilteredProjectListWithAllProjects[currentHighlightedIndex]
+      this.highlightedProjectName = newHighlightedProject.metadata.name
 
-      .project-filter {
-        align-items: center
+      if (currentHighlightedIndex >= this.numberOfVisibleProjects - 1) {
+        this.numberOfVisibleProjects++
       }
 
-      .project-add > div {
-        justify-content: left;
+      this.scrollSelectedProjectIntoView()
+    },
+    scrollSelectedProjectIntoView () {
+      const projectListChildren = get(this, '$refs.projectList.$children')
+      if (!projectListChildren) {
+        return
+      }
+      const projectListItem = find(projectListChildren, child => {
+        return get(child, '$vnode.data.key') === this.highlightedProject.metadata.name
```

Comment:
is this save? we access internal properties, but I've added a test for this method. We could also introduce an `id` attribute with identical value. WDYT?